src/main/java/org/embulk/input/jira/client/JiraClient.java

```diff
@@ -0,0 +1,197 @@
+                            })
+                            .collect(Collectors.toList());
+    }
+
+    public int getTotalCount(final PluginTask task)
+    {
+        String response = searchJiraAPI(task, 0, MIN_RESULTS);
+        JsonParser parser = new JsonParser();
+        JsonObject result = parser.parse(response).getAsJsonObject();
+        return result.get("total").getAsInt();
+    }
+
+    private String searchJiraAPI(final PluginTask task, int startAt, int maxResults)
+    {
+        try {
+            return retryExecutor().withRetryLimit(task.getRetryLimit())
+            .withInitialRetryWait(task.getInitialRetryIntervalMillis())
+            .withMaxRetryWait(task.getMaximumRetryIntervalMillis())
+            .runInterruptible(new Retryable<String>()
+            {
+                @Override
+                public String call() throws Exception
+                {
+                    return authorizeAndRequestWithGet(task, JiraUtil.buildSearchUrl(task, startAt, maxResults));
+                }
+
+                @Override
+                public boolean isRetryableException(Exception exception)
+                {
+                    if (exception instanceof JiraException) {
+                        int statusCode = ((JiraException) exception).getStatusCode();
+                        if (statusCode / 100 == 4 && statusCode != HttpStatus.SC_UNAUTHORIZED && statusCode != 429) {
+                            return false;
+                        }
+                        return true;
+                    }
+                    return false;
+                }
+
+                @Override
+                public void onRetry(Exception exception, int retryCount, int retryLimit, int retryWait)
+                        throws RetryGiveupException
+                {
+                    if (exception instanceof JiraException) {
+                        String message = String
+                                .format("Retrying %d/%d after %d seconds. HTTP status code: %s",
+                                        retryCount, retryLimit,
+                                        retryWait / 1000,
+                                        ((JiraException) exception).getStatusCode());
+                        LOGGER.warn(message);
+                    }
+                    else {
+                        String message = String
+                                .format("Retrying %d/%d after %d seconds. Message: %s",
+                                        retryCount, retryLimit,
+                                        retryWait / 1000,
+                                        exception.getMessage());
+                        LOGGER.warn(message, exception);
+                    }
+                }
+
+                @Override
+                public void onGiveup(Exception firstException, Exception lastException) throws RetryGiveupException
+                {
+                    LOGGER.warn("Retry Limits Completed");
```

Comment:
Code style: just wording, "Completed" sounds a little bit misleading, imo "Retry Limit Exceeded" would be easier to understand from the logs.

Reply:
Done